Go library examples in docs don't work

Repro is with Zed commit 7ba4ecc.

I'm not much of a Go developer, but I was hacking a bit and tried working off the user-facing docs. However, what's shown there doesn't currently work. Starting with what's shown for just the top [ZSON Reader example](https://zed.brimdata.io/docs/next/libraries/go#zson-reader):

```
$ mkdir example

$ cd example

$ go mod init example
go: creating new go.mod: module example

$ cat > main.go
# [pasted the example from the docs]

$ go mod tidy
go: finding module for package github.com/brimdata/zed/zson
go: finding module for package github.com/brimdata/zed
go: finding module for package github.com/brimdata/zed/zio/zsonio
go: found github.com/brimdata/zed in github.com/brimdata/zed v1.8.1
go: found github.com/brimdata/zed/zio/zsonio in github.com/brimdata/zed v1.8.1
go: found github.com/brimdata/zed/zson in github.com/brimdata/zed v1.8.1

$ echo '{s:"hello"}{x:123}{s:"world"}' | go run .
# example
./main.go:15:32: cannot use os.Stdin (variable of type *os.File) as *"github.com/brimdata/zed".Context value in argument to zsonio.NewReader
./main.go:15:42: cannot use zctx (variable of type *"github.com/brimdata/zed".Context) as io.Reader value in argument to zsonio.NewReader: *"github.com/brimdata/zed".Context does not implement io.Reader (missing method Read)
```

--- a/docs/libraries/go.md
+++ b/docs/libraries/go.md
@@ -55,6 +55,7 @@ import (
 func main() {
 	zctx := zed.NewContext()
 	reader := zsonio.NewReader(zctx, os.Stdin)
+	arena := zed.NewArena()
 	for {
 		val, err := reader.Read()
 		if err != nil {
@@ -63,9 +64,9 @@ func main() {
 		if val == nil {
 			return
 		}
-		s := val.Deref("s")
+		s := val.Deref(arena, "s")
 		if s == nil {
-			s = zctx.Missing().Ptr()
+			s = zctx.Missing(arena).Ptr()
 		}
 		fmt.Println(zson.String(s))
 	}
@@ -139,6 +140,7 @@ func main() {
 	}
 	defer reader.Close()
 	zctx := zed.NewContext()
+	arena := zed.NewArena()
 	for {
 		val, err := reader.Read()
 		if err != nil {
@@ -147,9 +149,9 @@ func main() {
 		if val == nil {
 			return
 		}
-		s := val.Deref("s")
+		s := val.Deref(arena, "s")
 		if s == nil {
-			s = zctx.Missing().Ptr()
+			s = zctx.Missing(arena).Ptr()
 		}
 		fmt.Println(zson.String(s))
 	}

--- a/docs/libraries/go.md
+++ b/docs/libraries/go.md
@@ -39,7 +39,7 @@ go get github.com/brimdata/zed
 ### ZSON Reader
 
 Read ZSON from stdin, dereference field `s`, and print results:
-```
+```mdtest-go-example
 package main
 
 import (
@@ -54,7 +54,7 @@ import (
 
 func main() {
 	zctx := zed.NewContext()
-	reader := zsonio.NewReader(os.Stdin, zctx)
+	reader := zsonio.NewReader(zctx, os.Stdin)
 	for {
 		val, err := reader.Read()
 		if err != nil {
@@ -65,7 +65,7 @@ func main() {
 		}
 		s := val.Deref("s")
 		if s == nil {
-			s = zctx.Missing()
+			s = zctx.Missing().Ptr()
 		}
 		fmt.Println(zson.String(s))
 	}
@@ -105,7 +105,7 @@ zed create -lake scratch Demo
 echo '{s:"hello, world"}{x:1}{s:"good bye"}' | zed load -lake scratch -use Demo -
 ```
 Now replace `main.go` with this code:
-```
+```mdtest-go-example
 package main
 
 import (
@@ -129,7 +129,7 @@ func main() {
 		log.Fatalln(err)
 	}
 	ctx := context.TODO()
-	lake, err := api.OpenLake(ctx, uri.String())
+	lake, err := api.OpenLake(ctx, nil, uri.String())
 	if err != nil {
 		log.Fatalln(err)
 	}
@@ -149,7 +149,7 @@ func main() {
 		}
 		s := val.Deref("s")
 		if s == nil {
-			s = zctx.Missing()
+			s = zctx.Missing().Ptr()
 		}
 		fmt.Println(zson.String(s))
 	}

--- a/mdtest/mdtest.go
+++ b/mdtest/mdtest.go
@@ -192,6 +192,11 @@ func parseMarkdown(source []byte) (map[string]string, []*Test, error) {
 				Line:     fcbLineNumber(commandFCB, source),
 			})
 			commandFCB = nil
+		case "mdtest-go-example":
+			tests = append(tests, &Test{
+				GoExample: fcbLines(fcb, source),
+				Line:      fcbLineNumber(fcb, source),
+			})
 		}
 		return ast.WalkContinue, nil
 	})

--- a/mdtest/test.go
+++ b/mdtest/test.go
@@ -3,7 +3,9 @@ package mdtest
 import (
 	"errors"
 	"fmt"
+	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 
 	"github.com/pmezard/go-difflib/difflib"
@@ -11,16 +13,20 @@ import (
 
 // Test represents a single test in a Markdown file.
 type Test struct {
-	Command  string
-	Dir      string
-	Expected string
-	Fails    bool
-	Head     bool
-	Line     int
+	Command   string
+	Dir       string
+	Expected  string
+	Fails     bool
+	Head      bool
+	Line      int
+	GoExample string
 }
 
 // Run runs the test, returning nil on success.
 func (t *Test) Run() error {
+	if t.GoExample != "" {
+		return t.vetGoExample()
+	}
 	c := exec.Command("bash", "-e", "-o", "pipefail")
 	c.Dir = t.Dir
 	c.Stdin = strings.NewReader(t.Command)
@@ -56,4 +62,22 @@ func (t *Test) Run() error {
 		return fmt.Errorf("expected and actual output differ:\n%s", diff)
 	}
 	return nil
+}
+
+func (t *Test) vetGoExample() error {
+	dir, err := os.MkdirTemp("", "")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(dir)
+	path := filepath.Join(dir, "main.go")
+	if err := os.WriteFile(path, []byte(t.GoExample), 0666); err != nil {
+		return err
+	}
+	_, err = exec.Command("go", "vet", path).Output()
+	var exitErr *exec.ExitError
+	if errors.As(err, &exitErr) {
+		return fmt.Errorf("could not vet go example: %s", string(exitErr.Stderr))
+	}
+	return err
 }

